xocto/settlement_periods.py

```diff
@@ -1,61 +1,130 @@
+import pytz
+import typing
 import datetime
 
-import pytz
+from . import exceptions
+
+
+__all__ = [
+    'convert_sp_and_date_to_utc',
+    'convert_utc_to_sp_and_date',
+    'number_of_periods_in_timedelta'
+]
+
+
+# Time zones implemented
+UTC_TZ = "UTC"
+GB_TZ = "Europe/London"
+
+
+def _to_timezone(dt: datetime.datetime, timezone: str) -> datetime.datetime:
+    """
+    Converts an aware datetime to another time zone.
+    """
+    tz = pytz.timezone(timezone)
+    # Rebasing to local time zone
+    local_dt = dt.astimezone(tz)
+    # Normalizing for daylight saving rules
+    return tz.normalize(local_dt)
 
 
-def _to_timezone(value, timezone):
+def _get_first_sp(d: datetime.date, timezone: str, is_wholesale: bool) -> datetime.datetime:
```

Comment:
I think we could make the module more readable by passing the actual tzinfo object around to these private functions, rather than passing the string version. It would avoid all the `pytz.timezone()` calls.